addons/mail/static/src/js/form_renderer.js

```diff
@@ -1,77 +1,152 @@
-            var updatedMailFields = _.intersection(fields, chatterFields);
-            if (updatedMailFields.length) {
-                this.chatter.update(state, updatedMailFields);
+    _renderNode(node) {
+        if (node.tag === 'div' && node.attrs.class === 'oe_chatter') {
+            this._formHasChatter = true;
+            return null;
+        } else {
+            return this._super(...arguments);
+        }
+    },
+
+    /**
+     * Overrides the function to render the chatter once the form view is rendered.
+     * @returns {Promise<void>}
+     * @private
+     */
+    async _renderView() {
+        await this._super(...arguments);
+        if (this._formHasChatter) {
+            Chatter.env = this.env;
+            if (this._chatter)
+            {
+                if (this._oldState.res_id !== this.state.res_id || this._oldState.model !== this.state.model)
+                {
+                    await this._deleteChatter();
+                    await this._createChatter();
+                } else {
+                    this._chatter.unmount();
+                    await this.env.store.dispatch('updateChatter', this._chatterLocalId);
+                    await this._mountChatter();
+                }
+            }  else {
+                await this._createChatter();
             }
         }
-        return this._super.apply(this, arguments);
     },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
 
     /**
-     * Overrides the function that renders the nodes to return the chatter's $el
-     * for the 'oe_chatter' div node.
-     *
-     * @override
+     * Create the chatter
+     * @returns {Promise<void>}
      * @private
      */
-    _renderNode: function (node) {
-        var self = this;
-        if (node.tag === 'div' && node.attrs.class === 'oe_chatter') {
-            if (!this.chatter) {
-                this.chatter = new Chatter(this, this.state, this.mailFields, {
-                    isEditable: this.activeActions.edit,
-                    viewType: 'form',
-                });
-
-                var $temporaryParentDiv = $('<div>');
-                this.defs.push(this.chatter.appendTo($temporaryParentDiv).then(function () {
-                    self.chatter.$el.unwrap();
-                    self._handleAttributes(self.chatter.$el, node);
-                }));
-                return $temporaryParentDiv;
-            } else {
-                this.chatter.update(this.state);
-                return this.chatter.$el;
-            }
-        } else {
-            return this._super.apply(this, arguments);
+    async _createChatter() {
+        // Generate chatter local id (+ fake thread or start loading real thread)
+        const chatterLocalId = this.env.store.dispatch('initChatter', {
+            id: this.state.res_id,
+            model: this.state.model,
+        });
```

Comment:
`createChatter`
would also name args `initialThreadId` and `initialThreadModel`, for clarity